src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1675,6 +1696,9 @@ trait Namers extends MethodSynthesis {
 
   abstract class TypeCompleter extends LazyType {
     val tree: Tree
+    override def forceDirectSuperclasses: Unit = {
+      tree.foreach(t => Option(t.symbol).map(_.maybeInitialize))
```

Comment:
@retronym any thoughts on the idea of using a flag to record that `knownDirectSubclasses` has been observed so that we can error if any additional subclasses add themselves later?